js/command.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as dispatch from "./dispatch";
+import { DenoError, ErrorKind } from "./errors";
+import * as flatbuffers from "./flatbuffers";
+import * as msg from "gen/msg_generated";
+import { assert, isObject, unreachable } from "./util";
+
+export interface CommandOptions {
+  argv?: string[];
+  dir?: string | null;
+  throw?: boolean;
```

Comment:
I am trying to avoid the node.js situation where there are like 10 different functions that all do something slightly different and they all take their options slightly different and return their results slightly different. This method can't run batch files and that one can't combine stdout+stderr and whatnot. That's super annoying and hard to remember even for someone who has been knee deep in their implementation. 

This options object was explicitly to avoid this situation.
I want to be able to specify such a convenience method. If we do add a bunch of convenience wrappers we should be able to say `this is the same as all the other method except here the defaults are `{ throw: false, stdio: ['null', 'buffered_string', '&'], tralala: 'dingdong' }`.

Reply:
Ok, makes sense.